tests/5.1/begin_end_declare_variant.c

```diff
@@ -18,65 +14,59 @@
 #include "ompvv.h"
 
 #define N 1024
-int arr[N]; // implicit map 3 variables 
-int errors = 0;
+int arr[N]; // implicit map array 
+int errors;
 int i = 0;
 
-
-void t_add(int *arr){
-	for (int i = 0; i < N; i++){
+void add(int *arr){
+	for (int i = 0; i < N; i++){ // Base function adds 1 to array values
 		arr[i] = i+1;
 	}
 }
 
-#pragma omp begin declare variant match( construct={parallel} ) // If inside a parallel for, should use multiply function instead of add
-	void t_add(int *arr){
+#pragma omp begin declare variant match( construct={parallel} ) // Parallel variant adds 2 to array values
+	void add(int *arr){
 		#pragma omp for
-		for (int i = 0; i < N; i ++){
+		for (int i = 0; i < N; i++){
 			arr[i] = i + 2;
-		}
+		} 
 	}
 #pragma omp end declare variant
 
-#pragma omp begin declare variant match( construct={target} ) // If inside target region
-	void t_add(int *arr){
+#pragma omp begin declare variant match( construct={target} ) // Target variant adds 3 to array values
+	void add(int *arr){
```

Comment:
should this be defined **in** a target region?

Reply:
add omp_is_initial_device() here to ensure that it is definitely running on the target